pkg/scheduler/podtask/ranges_test.go

```diff
@@ -0,0 +1,142 @@
+package podtask
+
+import (
+	"reflect"
+	"testing"
+
+	mesos "github.com/mesos/mesos-go/mesosproto"
+)
+
+func TestNewPortRanges(t *testing.T) {
+	t.Parallel()
+
+	for i, tt := range []struct {
+		Ranges
+		want Ranges
+	}{
+		{Ranges{{2, 0}, {3, 10}}, Ranges{{0, 10}}},
```

Comment:
The first case needs squashing due to overlap. I can add a case for squashing due to continuity.